bin/rucio

```diff
@@ -421,58 +430,62 @@ def list_file_replicas(args):
-                    table.append([replica['scope'], replica['name']])
-            print(tabulate(table, tablefmt=tablefmt, headers=['SCOPE', 'NAME']))
-        elif args.link:
-            pfn_dir, dst_dir = args.link.split(':')
+        return SUCCESS
+
+    if args.missing:
+        for replica, rse in itertools.product(replicas, rses):
+            if 'states' in replica and rse in replica['states'] and replica['states'].get(rse) != 'AVAILABLE':
+                table.append([replica['scope'], replica['name'], "({0}) {1}".format(ReplicaState[replica['states'].get(rse)].value, rse)])
+        print(tabulate(table, tablefmt=tablefmt, headers=['SCOPE', 'NAME', '(STATE) RSE']))
+    elif args.link:
+        pfn_dir, dst_dir = args.link.split(':')
+        if args.rses:
+            for replica, rse in itertools.product(replicas, rses):
+                if rse in list(replica['rses'].keys()) and replica['rses'][rse]:
+                    for pfn in replica['rses'][rse]:
+                        os.symlink(dst_dir + pfn.rsplit(pfn_dir)[-1], replica['name'])
+        else:
             for replica in replicas:
-                if args.selected_rse:
-                    if args.selected_rse in list(replica['rses'].keys()) and replica['rses'][args.selected_rse]:
-                        for pfn in replica['rses'][args.selected_rse]:
+                for rse in replica['rses']:
+                    if replica['rses'][rse]:
+                        for pfn in replica['rses'][rse]:
                             os.symlink(dst_dir + pfn.rsplit(pfn_dir)[-1], replica['name'])
-                else:
-                    for rse in replica['rses']:
-                        if replica['rses'][rse]:
-                            for pfn in replica['rses'][rse]:
-                                os.symlink(dst_dir + pfn.rsplit(pfn_dir)[-1], replica['name'])
-        elif args.pfns:
-            for replica in replicas:
-                if args.selected_rse:
-                    if args.selected_rse in list(replica['rses'].keys()) and replica['rses'][args.selected_rse]:
-                        for pfn in replica['rses'][args.selected_rse]:
-                            print(pfn)
-                else:
-                    for rse in replica['rses']:
-                        if replica['rses'][rse]:
-                            for pfn in replica['rses'][rse]:
-                                print(pfn)
+    elif args.pfns:
+        if args.rses:
+            for replica, rse in itertools.product(replicas, rses):
+                if rse in list(replica['rses'].keys()) and replica['rses'][rse]:
+                    for pfn in replica['rses'][rse]:
+                        print(pfn)
         else:
-            if args.all_states:
-                header = ['SCOPE', 'NAME', 'FILESIZE', 'ADLER32', '(STATE) RSE: REPLICA']
-            else:
-                header = ['SCOPE', 'NAME', 'FILESIZE', 'ADLER32', 'RSE: REPLICA']
             for replica in replicas:
```

Comment:
Is this a bug which is being fixed? If yes, would be nice to have a short sentence about it in the commit message.

Reply:
looks correct. I got confused by github diff. Sorry for the noise